packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -207,13 +301,23 @@ export default compose( [
 			title,
 			attributes,
 			blockType,
-			isFirstBlock,
 			isLastBlock,
 			isSelected,
 			isValid,
 			getAccessibilityLabelExtra,
 			showFloatingToolbar,
 			parentId,
+			isDashed,
+			isDimmed,
+			firstToSelectId,
+			isInnerBlock,
```

Comment:
It looks like ultimately `isInnerBlock` is passed down the hierarchy to end up in `RichText`, preventing touches there to avoid the OS making the text view focused. I don't think we can expect all blocks that use `RichText` to respect that convention, and it shouldn't be necessary.

What I'd suggest instead is that whenever we want to disable touches on the children blocks, instead of passing down this prop, we can set `pointerEvents='box-only` in the `TouchableWithoutFeedback` that's capturing the user's touch in `BlockListBlock.render`.

Reply:
The issue I had been dealing was connected with AztecRichTextView which catch focus just after click in text area. Adding `pointerEvents = "none"` solves that issue but produce app gets into circular setState after adding new RichText block when you were in text editing mode in other.

The solution with passing info about text block is child of other block ( isInnerBlock ) and add some logic inside RichText component helps to solve both.

I have checked solution you provide and it seems that it's not respected by RichText editor ( which gets the focus anyway just after onPress on block )

Reply:
This seems to be somewhat working: https://github.com/WordPress/gutenberg/commit/fdf77107ad4fc2b7c4cdd8ce7fb69268ff1dd65d

I'm seeing the issue you mention with the RichText selection loop, but I don't see why the `pointerEvents` configuration would cause this. It looks more like a different bug that was uncovered by this, and I think we should investigate what is actually causing the selection loop.

Reply:
`pointerEvents` just prevent `AztecView` to catch the `onPress` event. Selection loop issue was handle by introduce `isActive` in component state (initial value of it depends also on `isInnerBlock` passed from `block.native.js`)

However I agree that it can be a bug discovered in here and we might need further investigation.

Reply:
> Selection loop issue was handle by introduce isActive in component state (initial value of it depends also on isInnerBlock passed from block.native.js)

🤔 I had removed all of that in fdf7710 and was still seeing the selection loop

Reply:
@koke to make sure we are on the same page. `isActive` in state of `RichText` component do the job with selection loop issue.

Please check my comment [here](https://github.com/WordPress/gutenberg/commit/fdf77107ad4fc2b7c4cdd8ce7fb69268ff1dd65d#r35909772)

Reply:
But I don't understand why that would work. According to [documentation](https://facebook.github.io/react-native/docs/view.html#pointerevents), `pointerEvents`

> Controls whether the View can be the target of touch events.

All that `isActive` was doing in `RichText` was setting `pointerEvents: none` when not selected, which would prevent touches. But the loop isn't caused by touches since you only tap the screen once for it to happen, so it seems to me that this solution was masking a different problem.

Reply:
I agree that something weird happens here.

In `componentDidUpdate` there is a logic which calls `this._editor.focus()` and `this.onSelectionChange` depending of current and previous selection. 

Possible explanation is that the login in `componentDidMount` cause the selection loop. Its interesting that it goes to that loop only when you are in text editing mode so the conclusion is that it might be something with `this._editor.focus() / this.editor.blur()` and the `AztecView`

WDYT ?

Reply:
https://github.com/WordPress/gutenberg/commit/0f7d39d8b8a3a14239168a48245c00a82bac47d2 seems to fix it. I described this on slack, but copying it here for reference:

Before any of these changes, when a block is inserted a bunch of things happen, on React side you can see:
- Inserter closes, dismissing the modal
- Existing RichText gets deselected, componentDidUpdate tries to blur the text view
- New RichText componentDidMount focuses the text view
- New RichText gets selected, componentDidUpdate tries to focus the text view again

Now react-native batches updates and tries to do all those in order, which can be seen on the native debugger
- The modal is dismissed, which restores the focus to the old paragraph
- The new view gets focus
- The new view requests focus again

The key here is in that first step. In `RCTAztecView.becomeFirstResponder()`, isFirstResponder is true, so Aztec doesn’t try to notify the React side that it got the focus because it already had it.

However, in my approach in fdf77107ad4fc2b7c4cdd8ce7fb69268ff1dd65d, setting `pointer-events: none` in the block translates to setting `userInteractionEnabled = false`, which in turns tries to call `resignFirstResponder` on the current responder (the old paragraph). This means that when the modal is dismissed, the old paragraph would try to regain focus, and so it begins a discussion between the JS and native side over which RichText is actually selected.

Switching to `pointer-events: box-only` also disables touches on the children components but doesn't set `userInteractionEnabled`, avoiding this issue.

I think the underlying general issue is that we are trying to sync focus and selection state between native and JS, but updates happen in batches, which can lead to loops if things happen in the wrong order. I can't wait for that new react-native architecture to make our AztecView communication synchronous 😁